python/phonenumbers/phonenumberutil.pyi

```diff
@@ -0,0 +1,204 @@
+def region_codes_for_country_code(country_code: int) -> tuple[str, ...]: ...
+def country_code_for_region(region_code: str) -> int: ...
+def country_code_for_valid_region(region_code: str) -> int: ...
+def ndd_prefix_for_region(region_code: str, strip_non_digits: bool) -> str | None: ...
+def is_nanpa_country(region_code: str) -> bool: ...
+def is_alpha_number(number: str) -> bool: ...
+def is_possible_number(numobj: PhoneNumber) -> bool: ...
+def is_possible_number_for_type(numobj: PhoneNumber, numtype: int) -> bool: ...
+def _test_number_length(national_number: str, metadata: PhoneMetadata, numtype: int = ...) -> int: ...
+def is_possible_number_with_reason(numobj: PhoneNumber) -> int: ...
+def is_possible_number_for_type_with_reason(numobj: PhoneNumber, numtype: int) -> int: ...
+def is_possible_number_string(number: str, region_dialing_from: str) -> bool: ...
+def truncate_too_long_number(numobj: PhoneNumber) -> bool: ...
+def _extract_country_code(number: str) -> tuple[int, str]: ...
+def _maybe_extract_country_code(number: str, metadata: PhoneMetadata, keep_raw_input: bool, numobj: PhoneNumber) -> tuple[int, str]: ...
+def _parse_prefix_as_idd(idd_pattern: Pattern[str], number: str) -> tuple[bool, str]: ...
+def _maybe_strip_i18n_prefix_and_normalize(number: str, possible_idd_prefix: str) -> tuple[int, str]: ...
+def _maybe_strip_national_prefix_carrier_code(number: str, metadata: PhoneMetadata) -> tuple[str, str, bool]: ...
+def _maybe_strip_extension(number: str) -> tuple[str, str]: ...
+def _check_region_for_parsing(number: str | None, default_region: str | None) -> bool: ...
+def _set_italian_leading_zeros_for_phone_number(national_number: str, numobj: PhoneNumber) -> None: ...
+def parse(number: str, region: str | None = ..., keep_raw_input: bool = ..., numobj: PhoneNumber | None = ..., _check_region: bool = ...) -> PhoneNumber: ...
+def _build_national_number_for_parsing(number: str) -> str: ...
+def _copy_core_fields_only(inobj: PhoneNumber) -> PhoneNumber: ...
+def _is_number_match_OO(numobj1_in: PhoneNumber, numobj2_in: PhoneNumber) -> int: ...
+def _is_national_number_suffix_of_other(numobj1: PhoneNumber, numobj2: PhoneNumber) -> bool: ...
+def _is_number_match_SS(number1: str, number2: str) -> int: ...
+def _is_number_match_OS(numobj1: PhoneNumber, number2: str) -> int: ...
+def is_number_match(num1: PhoneNumber | str, num2: PhoneNumber | str) -> int: ...
+def can_be_internationally_dialled(numobj: PhoneNumber) -> bool: ...
+def is_mobile_number_portable_region(region_code: str) -> bool: ...
+
+class NumberParseException(UnicodeMixin, Exception):
+    INVALID_COUNTRY_CODE: int
+    NOT_A_NUMBER: int
+    TOO_SHORT_AFTER_IDD: int
+    TOO_SHORT_NSN: int
+    TOO_LONG: int
+    error_type: int
+    def __init__(self, error_type: int, msg: str) -> None: ...
+    def __reduce__(self) -> tuple[type[NumberParseException], tuple[int, str]]: ...
```

Comment:
Not sure what's the right thing here &ndash; the first thing in the returned `tuple` is [supposed](https://docs.python.org/3/library/pickle.html#object.__reduce__) to be a callable that can take as arguments the second thing in the tuple.  So should this be something like
```python
def __reduce__(self) -> tuple[Callable[[something], NumberParseException], tuple[int,str]]: ...
```

Reply:
I'm not sure here -- as `callable(NumberParseException)` is `True`, and this might be a case of purity beating specificity? Typeshed has a few examples of the first arg being a type, callable, or Any, so isn't very helpful.

If it were to be callable you'd want `Callable[[int, str], NumberParseException]` -- but I think as it currently is is probably better? Though not a big issue either way

Reply:
As-is seems fine.